src/language-js/needs-parens.js

```diff
@@ -215,7 +215,11 @@ function needsParens(path, options) {
           if (!parent) {
             return false;
           }
-          if (parent.type === "ForStatement" && parent.init === node) {
+          if (
+            parent.type === "ForStatement" &&
+            // eslint-disable-next-line prettier-internal-rules/better-parent-property-check-in-needs-parens
```

Comment:
How about renaming the `parent` variable to `ancestor` to get rid of `eslint-disable`?

Reply:
This function is not right

**Prettier 2.1.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAZgCnRCACVKPAIwEMAnASgG48B6OvAHgFoW8p8AHChAZwA6UITnJ5MkKHxh4A+ngC84zJUUA+PMCF4CWHPkIkKNeo1bse5fkIC+lFbVqVgNoSAA0ICFxipofZFAKcggAdwAFXilkEFIAG1DSAE8Az2JyUjAAazgYAGUeMEIAc2QYcgBXOE84AFtiOAATRqaAGVIoYorSYrgAMQhyWtIYX06Y0gqYCA8QAAsYWriAdTnUeD5CuDz+ddQAN3WkmLA+VLQpOHIYcIzi4eR0eL5qkAArPgAPACEM7Ny8qRanBWoQ4I9nq8Pp88iU4nAAIoVCDwCFxF6eSwvcgxMgNOKzLjkQgwZaoRowObIAAcAAZMSEXssMlwYkS4Nj9uDPABHZHwW7eAIoUh8FhQOBNJqzKx81BWW49B5IJ7o14vWqoMqVdVwxH88EqyGeGCkYhkilUpAAJhNGVQcRKAGEILVlSAOQBWWYVF4AFTNwtVGJA+yqAEkoC1YHkwMSfABBKN5GBJeFol42GxAA)
```sh
--parser babel
```

**Input:**
```jsx
if(foo in bar); // <-- no parens

for (const _ = (() => {
  if(foo in bar); // <-- parens
})(); ; ){}

```

**Output:**
```jsx
if (foo in bar); // <-- no parens

for (
  const _ = (() => {
    if ((foo in bar)); // <-- parens
  })();
  ;

) {}

```

Reply:
https://github.com/prettier/prettier/issues/907#issuecomment-284304321